modules/porous_flow/doc/content/modules/porous_flow/natural_convection.md

```diff
@@ -0,0 +1,94 @@
+
+!listing modules/porous_flow/examples/natural_convection/natural_convection.i block=BCs
+
+### Executioner setup
+
+Lastly, we will tell Moose the type of the problem and how we want it to perform or so call Executioner setup (shown in the code block below). For this problem, a transient solver is required.
+To save time and computational resources, "IterationAdaptiveDT" and "Adaptivity"
+were implemented. The former enables the time step to be increased if the solution converges and decreased if it does not. This will help save time if a large time step can be used and aid in convergence with a small time step is required [(Details can be found here)](https://mooseframework.inl.gov/source/timesteppers/IterationAdaptiveDT.html). Similarly, the latter enables the mesh to automatically adapt to optimize the computational resource usage and resolve the area with stiff gradients. The mesh will be coarsened at the regions with low error and refined at the critical areas (high error and stiff gradients) [(Details can be found here)](https://mooseframework.inl.gov/moose/syntax/Adaptivity/). For practice, users can try to disable them by putting the code block into comment and witnessing the difference in the solving time and solution.
+
+!listing modules/porous_flow/examples/natural_convection/natural_convection.i block=Executioner
+
+### Auxilary kernels and variables (optional)
+
+This section is optional since these kernels and variables do not affect the calculation of the desired variable. Instead, they were declared as part of learning and for completeness since in practice, it is uncommon that we only care about a single property. In this example, we want to look at the fluid density at the end of the simulation thus we will add an auxkernel and an auxvariable for it. It can be noticed that the term "execute_on = TIMESTEP_END" indicates this kernel will be activated to calculate the fluid density at the last time step thus this calculation does not affect the performance of other tasks.
+
+!listing modules/porous_flow/examples/natural_convection/natural_convection.i block=AuxVariables
+
+!listing modules/porous_flow/examples/natural_convection/natural_convection.i block=AuxKernels
+
+## Result
+
+The input file can be executed and the result is as follows:
+
+!media media/porous_flow/natural_convection_video.mp4 style=width:100%;margin-left:10px; caption=Spatial distribution of temperature with a test time of 2 years. id=Temp_vid
+
+It is obvious that the mesh in the boundary between hot and cold gas is much more refined as discussed and surrounding areas have coarser mesh.
+
+!media media/porous_flow/natural_convection_last_frame.png style=width:100%;margin-left:10px; caption=Spatial distribution of temperature at the end of 2 years with mesh grid displayed. id=Temp_vid
+
```

Comment:
You need a unique id here - at the moment it is the same as the id in the vide. Maybe call this one Temp_twoyears